sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -118,6 +118,9 @@ class SessionCatalog(
   }
 
   def dropDatabase(db: String, ignoreIfNotExists: Boolean, cascade: Boolean): Unit = {
+    if (db == "default") {
```

Comment:
You are asking a very great question! Actually, I am not very sure how it works. 

First, the current code base is not considering case sensitivity in database names. That part is missing. I think we should call `toLowerCase` for all the cases.

Second, I saw we have a function `formatTableName` for formatting table names based on the configuration of `CASE_SENSITIVE`. However, the underlying Hive metastore is not case sensitive. See the document: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL

My question is how to achieve the case sensitivity when using Hive metastore? also cc @yhuai